bundles/CoreBundle/Command/ThumbnailsImageCommand.php

```diff
@@ -149,54 +169,82 @@ protected function execute(InputInterface $input, OutputInterface $output)
-        );
-        $progress->start();
+                $items[] = serialize([
+                    'image_id' => $image->getId(),
+                    'thumbnail' => $thumbnailConfig
+                ]);
+            }
+        }
+        return $items;
+    }
 
-        for ($i = 0; $i < (ceil($total / $perLoop)); $i++) {
-            $list->setLimit($perLoop);
-            $list->setOffset($i * $perLoop);
-            $images = $list->load();
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function runSingleCommand(string $item, InputInterface $input, OutputInterface $output)
+    {
+        $item = unserialize($item);
 
-            foreach ($images as $image) {
-                if (!$image instanceof Asset\Image) {
-                    continue;
-                }
+        $image = Asset\Image::getById($item['image_id']);
+        if (!$image instanceof Asset\Image) {
+            return;
+        }
 
-                if ($input->getOption('force')) {
-                    foreach ($thumbnailConfigNames as $thumbnailConfigName) {
-                        $image->clearThumbnail($thumbnailConfigName);
-                    }
-                }
+        $thumbnail = $item['thumbnail'] ?? null;
+        if(!$thumbnail instanceof Asset\Image\Thumbnail\Config) {
+            return;
+        }
 
-                foreach ($thumbnailsToGenerate as $thumbnailConfig) {
-                    $thumbnail = $image->getThumbnail($thumbnailConfig);
+        $thumbnail = $image->getThumbnail($thumbnail);
 
-                    $progress->setMessage(
-                        sprintf(
-                            'generated thumbnail for image [%d] | file: %s',
-                            $image->getId(),
-                            $thumbnail->getPath(false)
-                        )
-                    );
 
-                    $progress->advance(1);
-                }
-            }
-            \Pimcore::collectGarbage();
+        if ($output->isVerbose()) {
+            $output->writeln(
+                sprintf(
+                    'generated thumbnail for image [%d] | file: %s',
+                    $image->getId(),
+                    $thumbnail->getPath(false)
+                )
+            );
+        } else {
+            $thumbnail->getPath(false);
         }
+    }
 
-        $progress->finish();
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function runAfterBatch(InputInterface $input, OutputInterface $output)
+    {
+        \Pimcore::collectGarbage();
+    }
 
-        $output->writeln('');
+    /**
+     * @inheritdoc
+     *
+     */
+    protected function getItemName(int $count): string
+    {
+        return 1 === $count ? 'thumbnail' : 'thumbnails';
+    }
 
-        return 0;
+    /**
+     * @return \Symfony\Component\DependencyInjection\ContainerInterface
+     */
+    private function getContainer()
```

Comment:
@dvesh3 for what purpose? Maybe a leftover? If we need it we should use DI and only inject the needed services.

Reply:
used by Parallelization trait...changed visibility to protected

> Why is the progressbar gone :-(
> Should be still possible by using `runAfterBatch($input, $output, $items)`

Progressbar is still there (from Parallelization) w/o generated thumbnail info and i believe, it is quite hard to show generated thumbnail in `runAfterBatch` as it does not get the updated items

Reply:
👍  thanks for clarifying & sorry! 😊